build_test/lib/src/record_logs.dart

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:build/src/builder/logging.dart';
+import 'package:matcher/matcher.dart';
+import 'package:logging/logging.dart';
+
+/// Executes [run] with a new [Logger], returning the resulting log records.
```

Comment:
Can you mention how this behaves when `run` does or doesn't return a Future?

Also mention that the returned Stream never closes.

Reply:
Done.